### PR TITLE
`@once`, `@can`

### PR DESCRIPTION
**Pull request by claytonrcarter**
This adds support for `@once`, `@can` and `@pushOnce`. It also adds more failing tests (sorry!) for #44.

### CODE REVIEW

File: src/lang/lexer.ts [hunk 1]
```diff
@@ -28,17 +28,20 @@ enum Mode {
...
     "for",
     "foreach",
     "forelse",
+    "guest",
+    "isset",
+    "once",
+    "prepend",
+    "push",
+    "pushOnce",
+    "unless",
```
Comment (claytonrcarter) [inline src/lang/lexer.ts:44]: I sorted this alphabetically while I was here.

Comment (claytonrcarter) [reply src/lang/lexer.ts:44]: ... while I was here adding `can`, `once` and `pushOnce`. All other entries should be the same as before, just reordered.

**Comment by ryangjchandler** [comment]
Awesome, thanks @claytonrcarter!

**Comment by claytonrcarter** [comment]
Thank you!

**Pull request merged**